fl4health/server/tabular_feature_alignment_server.py

```diff
@@ -0,0 +1,142 @@
+                feature_info_source = self.poll_clients_for_feature_info(timeout)
+            # If the server already has the feature info, then it simply sends it to the clients.
+            else:
+                log(
+                    INFO,
+                    "Features information source already specified. Sending to clients to perform feature alignment.",
+                )
+                feature_info_source = self.tab_features_info.to_json()
+
+            # the feature information is sent to clients through the config parameter.
+            self.config[FEATURE_INFO] = feature_info_source
+            self.format_info_gathered = True
+
+            self.strategy.on_fit_config_fn = partial(fit_config, self.config, self.format_info_gathered)
+
+            # Now the server waits until feature alignment is performed on the clients' side
+            # and subsequently requests the input and output dimensions, which are needed for initializing
+            # the global model.
+            input_dimension, output_dimension = self.poll_clients_for_dimension_info(timeout)
+            log(DEBUG, f"input dimension: {input_dimension}, output dimension: {output_dimension}")
+            self.strategy.initial_parameters = self.initialize_parameters(input_dimension, output_dimension)
+            self.initial_polls_complete = True
+
+        # Normal federated learning rounds commence after all clients' features
+        # are aligned and global model is initialized.
+        return super().fit(num_rounds=num_rounds, timeout=timeout)
+
+    def poll_clients_for_feature_info(self, timeout: Optional[float]) -> str:
+        log(INFO, "Feature information source unspecified. Polling clients for feature information.")
+        assert isinstance(self.strategy, BasicFedAvg)
+        client_instructions = self.strategy.configure_poll(server_round=1, client_manager=self._client_manager)
+        # Randomly select one client to obtain its feature information.
+        client_instructions_rand_sample = random.sample(population=client_instructions, k=1)
+        results, _ = poll_clients(
+            client_instructions=client_instructions_rand_sample, max_workers=self.max_workers, timeout=timeout
+        )
+
+        assert len(results) == 1
+        _, get_properties_res = results[0]
+        feature_info = str(get_properties_res.properties[FEATURE_INFO])
+        return feature_info
+
+    def poll_clients_for_dimension_info(self, timeout: Optional[float]) -> Tuple[int, int]:
+        log(INFO, "Waiting for Clients to align features and then polling for dimension information.")
+        assert isinstance(self.strategy, BasicFedAvg)
+        client_instructions = self.strategy.configure_poll(server_round=1, client_manager=self._client_manager)
+        # Since the features of all clients are aligned, we just select one client
+        # to obtain the input/output dimensions.
+
```

Comment:
Super minor, but the space between the comment above and this call to poll clients should be removed since the comment applies to the call I think?